### ISSUE TITLE
Add runtime hints for MessageSource's message.properties

### ISSUE THREAD

**Issue by beckermarc** [spring-projects/spring-framework#30738]
While building a native image for a Spring Boot application that uses MessageSource and `message.properties` files I noticed that those are not registered as resources during the native image compilation by default.

After explicitly adding them to a `resource-config.json` file they were available in the native image. However I would have expected Spring Boot to provide a `RuntimeHintsRegistrar` implementation out-of-the-box that contributes these properties files automatically.

Is it reasonable to add such a `RuntimeHintsRegistrar` implementation?

**Comment by sdeleuze** [comment]
@philwebb @bclozel I am not sure to understand how we can handle this on Spring Framework side because, at AOT stage, I don't think we have access to `MessageSource` bean instances, and the basename is not available in the bean definition, but hidden in `org.springframework.boot.autoconfigure.context.MessageSourceAutoConfiguration#messageSource` `BeanInstanceSupplier`.

As far as I can tell, the only chance we have is to introduce the relevant hints (something like `resourceHints.registerPattern(baseName + ".properties").registerPattern(baseName + ".xml");` at Boot level based on the message source  configuration property.

Any thoughts?

**Comment by philwebb** [comment]
No, I think you're right. I was originally thinking that we could inspect the `ResourceBundleMessageSource`, but I think it's too risky to initialize that bean early.

Sorry @bclozel, could you transfer this back again?
